packages/api-server/src/routes/user/email-updates/get-email-update-setting.ts

```diff
@@ -18,10 +18,14 @@ const schema = {
 
 const root: FastifyPluginAsyncJsonSchemaToTs = async (app, opts): Promise<void> => {
     app.get('/', {
-        preHandler: [app.authenticate],
+        // @ts-ignore
+        preValidation: app.authenticate,
         schema
     }, async function (req, reply) {
-        const userId = req.user.id;
+        const { sub } = req.user as {
+          sub: string;
+        };
+        const userId = await app.userService.findOrCreateUserByAuth0Sub(sub, req.headers.authorization);
```

Comment:
Can we setup the userId in the jwt payload, do not fetch it from database every request?